packages/ethereum/contracts/HoprNetworkRegistry.sol

```diff
@@ -70,51 +87,82 @@ contract HoprNetworkRegistry is Ownable {
 
   /**
    * @dev Checks if the msg.sender fulfills registration requirement at the calling time, if so,
-   * register the EOA with HOPR node address. Account can also update its registration status
+   * register the EOA with HOPR node peer id. Account can also update its registration status
    * with this function.
-   * @notice It allows msg.sender to update registered node address.
-   * @param hoprAddress Hopr nodes id. e.g. 16Uiu2HAmHsB2c2puugVuuErRzLm9NZfceainZpkxqJMR6qGsf1x1
+   * @notice It allows msg.sender to update registered node peer id.
+   * @param hoprPeerId Hopr nodes peer id in bytes. e.g. 16Uiu2HAmHsB2c2puugVuuErRzLm9NZfceainZpkxqJMR6qGsf1x1
+   * hopr node peer id should always start with '16Uiu2HA' (0x3136556975324841) and be of length 53
    */
-  function selfRegister(string calldata hoprAddress) external mustBeEnabled returns (bool) {
+  function selfRegister(string calldata hoprPeerId) external mustBeEnabled returns (bool) {
+    require(
+      bytes(hoprPeerId).length == 53 && bytes32(bytes(hoprPeerId)[0:8]) == '16Uiu2HA',
```

Comment:
```suggestion
      bytes(hoprPeerId).length == 53,
```
Checking for that specific prefix is probably too much overfitting

Reply:
If the prefix is correct, can it be left there? Like suggested by @tolbrino, it can prevent accidentally wrongly registered entries.

Reply:
Checked the prefix and it indeed works